ethereum/helpers/uris.ts

```diff
@@ -47,27 +47,26 @@ export function getURITemplate(
           'thumb','ipfs://'||renders_cid||'/'||rig_id||'/'||image_thumb_name,
           'thumb_alpha','ipfs://'||renders_cid||'/'||rig_id||'/'||image_thumb_alpha_name,
           'animation_url',animation_base_url||rig_id||'.html',
-          'attributes',json_insert(
-            (
-              select json_group_array(
-                json_object('display_type',display_type,'trait_type',trait_type,'value',value)
-              )
-              from ${attributesTable} where rig_id=ID group by rig_id
-            ),
-            '$[#]',
-            json_object(
-              'display_type','string',
-              'trait_type','Garage Status',
-              'value',coalesce(
-                (select coalesce(end_time, 'in-flight') from ${pilotsTable} where rig_id=ID and end_time is null),
-                'parked'
-              )
-            )
+          'attributes', json_group_array(
+            json_object('display_type',display_type,'trait_type',trait_type,'value',value)
           )
-        ) from ${attributesTable} join ${lookupsTable} where rig_id=ID group by rig_id;`.replace(
-          /(\r\n|\n|\r|\s\s+)/gm,
-          ""
         )
+        from (
+          select *
+          from ${attributesTable}
+          union
+          select
+            a.rig_id,
+            'string' display_type,
+            'Garage Status' trait_type,
+            case when start_time is null then 'parked' else 'in-flight' end value
+          from
+            ${attributesTable} a
+            left join (select * from ${pilotsTable} where end_time is null) s on a.rig_id = s.rig_id
+        )
```

Comment:
This part of the `from` clause in between this set of parentheses has one goal: Select data from any/all tables you want to create metadata from, and combine them into a single virtual dataset by using `union`. So each `union`ed query must return the same columns that work with our json functions further above: `rig_id integer, display_type text, trait_type text, value text`. In this case we're just `union`ing the attributes table with data from the pilot sessions table, but it will be easy to add another `union` with data from any new table, like Filecoin deals, for example.